client/src/featureform/client.py

```diff
@@ -0,0 +1,75 @@
+from typing import Union
+from .register import (
+    ResourceClient,
+    SourceRegistrar,
+    LocalSource,
+    SubscriptableTransformation,
+)
+from .serving import ServingClient
+
+
+class Client(ResourceClient, ServingClient):
+    """
+    Client for interacting with Featureform APIs (resources and serving)
+
+    **Using the Client:**
+    ```py title="definitions.py"
+    import featureform as ff
+    from featureform import Client
+
+    client = Client("http://localhost:8080")
+
+    # Example 1: Get a registered provider
+    redis = client.get_provider("redis-quickstart")
+
+    # Example 2: Compute a dataframe from a registered source
+    transactions_df = client.compute_df("transactions", "quickstart")
+    """
+
+    def __init__(
+        self, host=None, local=False, insecure=False, cert_path=None, dry_run=False
+    ):
+        ResourceClient.__init__(
+            self,
+            host=host,
+            local=local,
+            insecure=insecure,
+            cert_path=cert_path,
+            dry_run=dry_run,
+        )
+        # Given both ResourceClient and ServingClient are instantiated together, if dry_run is True, then
+        # the ServingClient cannot be instantiated due to a conflict the local and host arguments.
+        if not dry_run:
+            ServingClient.__init__(
+                self, host=host, local=local, insecure=insecure, cert_path=cert_path
+            )
+
+    def compute_df(
```

Comment:
I know this is a change up about what I said before but I think we name this datafarme( instead of compute_df, the logic being:
to get a training set you'd call: client.training_set, it follows that for a dataframe you'd call: client.dataframe(